hack/generator/pkg/codegen/pipeline_determine_resource_ownership.go

```diff
@@ -0,0 +1,290 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package codegen
+
+import (
+	"context"
+	"github.com/Azure/k8s-infra/hack/generator/pkg/astmodel"
+	"github.com/pkg/errors"
+	"strings"
+)
+
+const resourcesPropertyName = astmodel.PropertyName("Resources")
+
+func determineResourceOwnership() PipelineStage {
+	return PipelineStage{
+		"Determine ARM resource relationships",
+		func(ctx context.Context, definitions astmodel.Types) (astmodel.Types, error) {
```

Comment:
You could just have `determineOwnership` take the context as well and get rid of this lambda. (I'm not sure how long this phase takes, maybe it's worth checking for cancellation in `determineOwnership`? I assume it's pretty quick though.)

Reply:
It's pretty fast. It looks like most pipeline stages don't actually do much with the ctx (I checked a few others and none of them were checking for cancellation).

I've made this change though - saves a few lines of code.

Reply:
Right, I think it mostly makes sense to check `ctx` when doing IO.